fabric-dynamic-registry-api-v1/src/main/java/net/fabricmc/fabric/api/dynamicregistry/v1/DynamicRegistryEntry.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.dynamicregistry.v1;
+
+import java.util.function.Supplier;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.Lifecycle;
+
+import net.minecraft.util.registry.DynamicRegistryManager;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.util.registry.SimpleRegistry;
+
+public class DynamicRegistryEntry<T> {
```

Comment:
Maybe just call this `DynamicRegistry` or `DynamicRegistryDefinition`, the `Entry` is more confusing than helpful

Reply:
Would `CustomDynamicRegistry` be good?

Reply:
Yeah, I came up with that name in my mind as well